src/standard_library/array_methods.md

```diff
@@ -0,0 +1,139 @@
+
+# Array
+
+For convenience, the STD provides some ready-to-use, common methods for arrays[^migrationNote]:
+
+## len
+
+Returns the length of an array
+
+```rust
+fn len(_array: Self) -> comptime Field
```

Comment:
We should remove `Self` in these docs. For example, `len` should be:

```rust
fn len<T, N>(_array: [T; N]) -> comptime Field
```

Since the `T, N` generics come from the impl as well.